Code/src/main/java/nl/utwente/group10/ui/components/blocks/Block.java

```diff
@@ -70,4 +77,12 @@ public final OutputAnchor getOutputAnchor() {
      * @return an expression that evaluates to what this block is.
      */
     public abstract Expr asExpr();
+    
+    /**
+     * This method should be called whenever the state of the Block has changed,
+     * the Update method will then call asExpr to evaluate the new state.
+     * Always use Update instead of asExpr unless you are sure you know what you
+     * are doing.
+     */
```

Comment:
Óf je maakt asExpr private, óf je zorgt ervoor dat ze beide probleemloos aangeroepen worden; vage dreigementen in comments heeft niemand wat aan :grin: